### PR TITLE
add: exposure status for sca findings in sarif output

### PR DESCRIPTION
**Pull request by brendongo**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by mmcqd — APPROVED** [review]
Should we include this in the JSON output too? Looks good tho, once you merge the snapshot changes

File: cli/src/semgrep/formatter/sarif.py [hunk 1]
```diff
@@ -217,6 +217,24 @@ def _rule_match_to_sarif(
...
+            elif rule_match.metadata.get("sca-kind") == "legacy":
+                exposure = "undetermined"
+            else:
+                # rule_match.metadata.get("sca-kind") == "reachable":
+                if rule_match.extra["sca_info"].reachable:
+                    exposure = "reachable"
+                else:
+                    exposure = "unreachable"
+
+            rule_match_sarif["properties"] = {"exposure": exposure}
```
Comment (mmcqd) [inline cli/src/semgrep/formatter/sarif.py:236]: Is this not overriding other properties?

Comment (brendongo) [reply cli/src/semgrep/formatter/sarif.py:236]: There aren't any right now but good call

Comment (brendongo) [reply cli/src/semgrep/formatter/sarif.py:236]: eh it will look ugly :D i think it's fine for now and test will catch if someone removed overrides this

**Pull request merged**